ledger/src/advance.rs

```diff
@@ -127,26 +127,48 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
                 let coinbase_verifying_key = self.coinbase_puzzle.coinbase_verifying_key();
                 // Retrieve the latest epoch challenge.
                 let latest_epoch_challenge = self.latest_epoch_challenge()?;
-                // TODO: For mainnet - Add `aborted_solution_ids` to the block. And optimize this logic.
-                // Verify the candidate solutions.
-                let verification_results: Vec<_> = cfg_into_iter!(candidate_solutions)
-                    .map(|solution| {
-                        (
-                            solution,
-                            solution
-                                .verify(coinbase_verifying_key, &latest_epoch_challenge, self.latest_proof_target())
-                                .unwrap_or(false),
-                        )
-                    })
-                    .collect();
+                // TODO: For mainnet - Add `aborted_solution_ids` to the block.
                 // Separate the candidate solutions into valid and aborted solutions.
                 let mut valid_candidate_solutions = Vec::with_capacity(N::MAX_SOLUTIONS);
                 let mut aborted_candidate_solutions = Vec::new();
-                for (solution, is_valid) in verification_results.into_iter() {
-                    if is_valid && valid_candidate_solutions.len() < N::MAX_SOLUTIONS {
-                        valid_candidate_solutions.push(solution);
+                // Reverse the candidate solutions in order to be able to chunk them more efficiently.
```

Comment:
```suggestion
                // Reverse the candidate solutions in order to be able to chunk them more efficiently.
                // Note: Reversing the order here preserves FIFO processing across rounds.
```
@ljedrz Can you confirm this has no impact to ordering from Narwhal across rounds?